rmf_websocket/include/rmf_websocket/BroadcastServer.hpp

```diff
@@ -59,6 +61,26 @@ class BroadcastServer
     ApiMessageCallback callback,
     std::optional<ApiMsgType> msg_selection = std::nullopt);
 
+  /// Add a callback to convert from a Description into an active Task.
+  ///
+  /// \param[in] port
+  ///   server url port number
+  ///
+  /// \param[in] callback
+  ///   callback function when the message is received
+  ///
+  /// \param[in] msg_selection
+  ///   selected msg type to listen. Will listen to all msg if nullopt
+  ///
+  /// \param[in] node_logging_interface
+  ///   node for logging
+  static std::shared_ptr<BroadcastServer> make_with_logger(
+    const int port,
+    ApiMessageCallback callback,
+    std::optional<ApiMsgType> msg_selection = std::nullopt,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr
+      node_logging_interface = nullptr);
```

Comment:
```suggestion
  static std::shared_ptr<BroadcastServer> make_with_logger(
    const int port,
    ApiMessageCallback callback,
    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr
      node_logging_interface,
    std::optional<ApiMsgType> msg_selection = std::nullopt);
```

On second thought, since we are introducing a new API, we should avoid the default behavior (if optional arguments are not provided) to be the same as the original API. I think it's a better idea to ensure that this API is specific for introducing the logging interface as a non-optional argument